tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2732,6 +2755,23 @@ SpirvVariable *DeclResultIdMapper::createSpirvStageVar(
     return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FullyCoveredEXT,
                                          isPrecise, srcLoc);
   }
+  // According to DXIL spec, the ShadingRate SV can only be used by GSOut,
+  // VSOut, or PSIn. According to Vulkan spec, the FragSizeEXT BuiltIn can only
+  // be used as PSIn.
+  case hlsl::Semantic::Kind::ShadingRate: {
+    switch (sigPointKind) {
+    case hlsl::SigPoint::Kind::PSIn:
+      stageVar->setIsSpirvBuiltin();
+      return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FragSizeEXT,
+                                           isPrecise, srcLoc);
+    default:
+      emitError("semantic ShadingRate currently unsupported in non-PS shader"
+                "stages",
+                srcLoc);
```

Comment:
hmm.. this is formatted somewhat weird. Have you run clang-format?

Reply:
Visual Studio did this when I saved the file. I'm assuming it's correct in doing this :)